app/src/main/java/com/kamwithk/ankiconnectandroid/ankidroid_api/NoteAPI.java

```diff
@@ -73,17 +59,33 @@ public boolean updateNoteFields(long note_id, String[] fields) throws Exception
         return api.updateNoteFields(note_id, fields);
     }
 
-    public long getNoteModelId(long note_id) throws Exception {
-        api.getNote(note_id); //set cursor to correct position
-        return api.getCurrentModelId();
+    public Long getNoteModelId(long note_id) {
+        // Manually queries the note with a specific projection to get the model ID
+        // Code copied/pasted from getNote() in AddContentAPI:
```

Comment:
Once again ideally we avoid this, but it's ok if it is unavoidable or a massive hassle to deal with

Reply:
Same function as in #42